src/tools/brush/ScissorsTool.js

```diff
@@ -0,0 +1,329 @@
+   * @param {(CornerstoneTools.event#TOUCH_DRAG|CornerstoneTools.event#MOUSE_DRAG|CornerstoneTools.event#MOUSE_MOVE)} evt  Interaction event emitted by an enabledElement
+   * @returns {void}
+   */
+  _setHandlesAndUpdate(evt) {
+    const eventData = evt.detail;
+    const element = evt.detail.element;
+
+    this._addPoint(eventData);
+    external.cornerstone.updateImage(element);
+  }
+
+  /**
+   * Function responsible for changing the Cursor, according to the strategy
+   * @param {HTMLElement} element
+   * @param {string} strategy The strategy to be used on Tool
+   * @private
+   * @returns {void}
+   */
+  _changeCursor(element, strategy) {
+    // Necessary to avoid setToolCursor without elements, what throws an error
+    if (!this.element) {
+      return;
+    }
+
+    const cursorList = {
+      FILL_INSIDE: scissorsFillInsideCursor,
+      FILL_OUTSIDE: scissorsFillOutsideCursor,
+      ERASE_OUTSIDE: scissorsEraseOutsideCursor,
+      ERASE_INSIDE: scissorsEraseInsideCursor,
+      default: scissorsFillInsideCursor,
+    };
+
+    const newCursor = cursorList[strategy] || cursorList.default;
+
+    setToolCursor(element, newCursor);
+    external.cornerstone.updateImage(element);
+  }
+
+  /**
+   * Event handler for MOUSE_UP/TOUCH_END during handle drag event loop.
+   *
+   * @private
+   * @method _applyStrategy
+   * @param {(CornerstoneTools.event#MOUSE_UP|CornerstoneTools.event#TOUCH_END)} evt Interaction event emitted by an enabledElement
+   * @returns {void}
+   */
+  _applyStrategy(evt) {
+    this._applySegmentationChanges(evt);
+    this._resetHandles();
+    external.cornerstone.updateImage(evt.detail.element);
+  }
+
+  /**
+   * Sets the start and end handle points to empty objects
+   *
+   * @private
+   * @method _resetHandles
+   * @returns {undefined}
+   */
+  _resetHandles() {
+    this.handles = {
+      points: [],
+    };
+
+    this.currentHandle = 0;
+  }
+
+  /**
+   * Adds a point on mouse click in polygon mode.
+   *
+   * @private
+   * @param {Object} eventData - data object associated with an event.
+   * @returns {undefined}
+   */
+  _addPoint(eventData) {
+    // If this is not the first handle
+    if (this.handles.points.length) {
+      // Add the line from the current handle to the new handle
+      this.handles.points[this.currentHandle - 1].lines.push({
+        x: eventData.currentPoints.image.x,
+        y: eventData.currentPoints.image.y,
+        lines: [],
+      });
+    }
+
+    // Add the new handle
+    this.handles.points.push({
+      x: eventData.currentPoints.image.x,
+      y: eventData.currentPoints.image.y,
+      lines: [],
+    });
+
+    // Increment the current handle value
+    this.currentHandle += 1;
+
+    // Force onImageRendered to fire
+    external.cornerstone.updateImage(eventData.element);
+  }
+
+  /**
+   * Change Strategy Method
+   * @param { string } strategy
+   * @private
+   * @returns {void}
+   */
+  _changeStrategy(strategy = 'default') {
```

Comment:
This should not have an underscore as it appears to be intended to be public API, at least from your demo.

Since this seems to be effectively a setter, I guess you could require this as as a `set` for the class.

Something like:
```js
  set activeStrategy (strategy) {
    this.activeStrategy = strategy;
    this._changeCursor(this.element, strategy);
    this._resetHandles();
  }
```

this way, the api for changing the activeStrategy remains the same as all other tools, i.e. `scissorToolInstance.activeStrategy`

@dannyrb maybe we should instead add a `setActiveStrategy` method in `BaseTool` that can be overridden to generalize this:

```js
/**
 * Change the active strategy.
 *
 * @virtual
 * @param  {string} strategy
 * @returns {null}
 */
setActiveStrategy(strategy) {
  this.activeStrategy = strategy;
}
```

Or can you have virtual ES6 setters? 🤔 

Sorry to hijack your PR for architecture discussion haha 😛 .

Reply:
@dannyrb we added `setActiveStrategy` to BaseTool, please, check if we can proceed this way. Seem to be the most generic and better, once this method can be used in every tool that needs to live change the cursor.